### ISSUE TITLE
Research: Learning System Implementation - What to Research

### ISSUE THREAD

**Issue by pselamy** [tradestreamhq/tradestream#1482]
## Parent Epic
#1454

## Overview

The learning system is a critical differentiator. Before implementing, we need systematic research on:
- Context-dependent Top-K selection
- Feedback loop architectures
- Performance attribution
- Self-improvement mechanisms

---

### 1. Top-K Selection in Context

**Problem**: "Top-K" means different things in different contexts:

| Context | What is "K"? | Considerations |
|---------|-------------|----------------|
| Strategy Selection | Best N strategies for current market | Regime detection, diversity, correlation |
| Few-Shot Examples | N examples for LLM spec generation | Quality > quantity, diverse patterns |
| Indicator Selection | Top N indicators for a strategy | Multicollinearity, information overlap |
| Parameter Space | Top N parameter sets per strategy | Exploration vs exploitation |
| Historical Performance | Last N results to consider | Recency bias vs sample size |

**Research Needed**:
- [ ] Literature review: Top-K in trading systems (Kelly Criterion, portfolio optimization)
- [ ] Literature review: Top-K in LLM few-shot learning (in-context learning papers)
- [ ] A/B testing framework for K selection
- [ ] Adaptive K based on confidence/variance

### 2. Feedback Loop Architecture

**Question**: How should performance data flow back into the system?

```
Current State:
┌────────────┐     ┌────────────┐     ┌────────────┐
│ Strategy   │────▶│  Backtest  │────▶│  Metrics   │
│ Discovery  │     │            │     │  (stored)  │
└────────────┘     └────────────┘     └────────────┘
         ▲                                    │
         │                                    │
         └────────── ??? ─────────────────────┘
```

**Options to Research**:
1. **Reinforcement Learning Loop** - Reward signal shapes discovery
2. **Evolutionary Pressure** - High performers spawn variants
3. **Attention Weighting** - LLM attends more to successful patterns
4. **Curriculum Learning** - Progressive difficulty/complexity

**Research Needed**:
- [ ] RL in trading literature (AlphaGo for trading?)
- [ ] Evolutionary strategies in quantitative finance
- [ ] Meta-learning approaches (learning to learn)
- [ ] Bandit algorithms for strategy selection

### 3. Performance Attribution

**Question**: When a strategy performs well, WHY did it work?

**Attribution Levels**:
1. **Indicator Attribution** - Which indicator contributed most?
2. **Parameter Sensitivity** - Which parameters mattered?
3. **Market Regime Attribution** - Did market conditions drive performance?
4. **Temporal Attribution** - Performance stable over time?

**Research Needed**:
- [ ] SHAP values for trading strategies
- [ ] Sensitivity analysis methods
- [ ] Regime detection algorithms
- [ ] Performance decomposition techniques

### 4. Self-Improvement Mechanisms

**Question**: How does the system get better autonomously?

```
Naive Loop:                    Sophisticated Loop:
                               
Strategy → Backtest → Store    Strategy → Backtest → Analyze → 
     ↑__________↓                   ↑____________________________↓
                                         │
                                         ▼
                               Identify failure modes →
                               Generate hypotheses →
                               Design experiments →
                               Test → Update beliefs
```

**Research Needed**:
- [ ] AutoML literature (hyperparameter optimization)
- [ ] Active learning for strategy space exploration
- [ ] Bayesian optimization for parameter tuning
- [ ] Monte Carlo Tree Search for strategy composition

### 5. Few-Shot Learning for Strategy Generation

**Question**: How do we select the best examples for LLM prompts?

**Considerations**:
- Diversity of examples (not all RSI-based)
- Quality threshold (only top performers?)
- Negative examples (what NOT to do?)
- Recency (recent successes vs all-time?)

**Research Needed**:
- [ ] In-context learning papers (GPT-3, Claude, etc.)
- [ ] Example selection for code generation
- [ ] Curriculum learning in LLM fine-tuning
- [ ] Chain-of-thought prompting for trading

### 6. Avoiding Overfitting to History

**Question**: How do we prevent the learning system from overfitting?

**Risks**:
- Strategies that worked in backtests fail live
- System learns to game backtest metrics
- Survivorship bias in top performers
- Regime changes invalidate historical patterns

**Research Needed**:
- [ ] Walk-forward optimization literature
- [ ] Out-of-sample testing best practices
- [ ] Cross-validation for time series
- [ ] Monte Carlo simulation for robustness

---

### Task 1: Literature Review - Learning in Trading Systems
- Academic papers on adaptive trading systems
- Papers on regime-switching models
- Machine learning in quantitative finance

### Task 2: Top-K Algorithm Survey
- Bandit algorithms (UCB, Thompson Sampling)
- Portfolio optimization (Markowitz, Kelly)
- Ensemble selection methods

### Task 3: Feedback Architecture Patterns
- Reinforcement learning architectures
- Evolutionary optimization patterns
- Meta-learning frameworks

### Task 4: LLM Few-Shot Selection
- In-context learning research
- Example selection algorithms
- Retrieval-augmented generation

---

### Approach A: Multi-Armed Bandit for Strategy Selection
```python
# Thompson Sampling for strategy selection
def select_strategy(strategies):
    samples = [beta.rvs(s.successes + 1, s.failures + 1) for s in strategies]
    return strategies[argmax(samples)]
```

### Approach B: Evolutionary + LLM Hybrid
```
1. GA optimizes parameters → quantitative
2. LLM generates new specs → creative
3. Combine: LLM proposes, GA refines
```

### Approach C: Retrieval-Augmented Spec Generation
```
1. User describes market hypothesis
2. Retrieve similar successful strategies
3. LLM synthesizes new spec from examples
```

---

## Dependencies
- #1473 (Learning Engine Core)
- #1463 (LLM Spec Generation)
- #1462 (Config-Based Strategies)

## Outputs Expected
1. Research summary document
2. Recommended approach with justification
3. Implementation plan with milestones
4. Evaluation metrics for learning effectiveness

---

## Labels
`research`, `learning-engine`, `priority-high`
